drivers/issi/is31fl3741.c

```diff
@@ -78,22 +78,6 @@ bool    g_scaling_registers_update_required[DRIVER_COUNT] = {false};
 
 uint8_t g_scaling_registers[DRIVER_COUNT][ISSI_MAX_LEDS];
```

Comment:
What I think you need to do here is have two sets of scaling/pwm register arrays, one for each page, and set them based on whether the r/g/b parts of `is31_led` has the 0x100 bit set (and mask it off beforehand).

Reply:
> What I think you need to do here is have two sets of scaling/pwm register arrays, one for each page, and set them based on whether the r/g/b parts of `is31_led` has the 0x100 bit set (and mask it off beforehand).

I can't get your point. why two sets?

Reply:
Because eg. the PWM register address of CS1,SW1 is 0x00 on page 0, but CS1,SW7 is also 0x00, but on page 1. So you either need two arrays for each page, or use some logic to merge the two together.

Reply:
we have two arrays(the size is 351, including page 0 and page 1) I think the easy way is using the index to distinguish the two pages.

Reply:
What if someone implements an RGB matrix but skips over some spots on the first page? Is that possible? In that case, distinguishing based on the index in `g_is31_leds` might be off.

Reply:
The ideal solution is write the reg of led which are need to be changed. if then the issi chip respond will be very slower than write all regs. so you can check the code of 3733, 3731, 3737 etc. Please check it!

Reply:
> What if someone implements an RGB matrix but skips over some spots on the first page? Is that possible? In that case, distinguishing based on the index of the LED in `g_is31_leds` might be off.

for the case, missing some spots will not happen, cause we keep all of regs of RGBs in the array and write the value for every LEDs at once. so don't worry about that.

Reply:
> 
> 
> I resolved the other review because I realised the CSx_SWx notation is better, and should probably be ported to the other drivers too.
> This review is focused on the fact that the registers are split across two pages and thus their values overlap. So there is the question of how to distinguish between eg. `CS1_SW1` and `CS1_SW7` because they are both `0x00`, but on different pages. My thought was to add `0x100` to the ones on the second page, and check that bit when deciding which page to use in the logic.
> Additionally, if we are to use these values as indexes into the `g_pwm_buffer` and `g_scaling_registers` arrays, there will be a gap in the middle, because the last value on page 0 is 0xB3, and the first on page 1 would be `0x100`. This seems inefficient, and I think that warrants splitting these arrays into two, for each page. The problem is, is it possible to use non-consecutive registers? If so, then using `g_is31_leds` as the index into the register arrays won't work.

The last index on page 0 is 0xB3, and the first index on page 1 is 0xB4. the address of page 1 register need add 0xB4.
and the two pages are continuous in the `g_pwm_buffer` and `g_scaling_registers` arrays.  You can check the attach file. 

We can distinguish the two pages by the index. if the index less than or equal to 0xB3 that will be page 0, if the index greater than 0xB3 that will be page 1. 


![pwm_register_20200824](https://user-images.githubusercontent.com/65591833/90997218-80446100-e5f3-11ea-9d42-48705901e145.png)

Reply:
@melgeek001365 's solution makes sense, no point using multiple arrays when they're effectively mapped the same way anyway. If anything, it's probably more logical to map them to the LEDs this way.

Reply:
How are the `CSx_SWx` defines mapped to these registers though? I don't see any logic here to test for `define & 0x100` to determine which page the register is on.
Right now, it doesn't look like this code actually works.

Reply:
See https://github.com/qmk/qmk_firmware/blob/5fa3e25aa320b3817577c6a1f9f43dd0a91b7072/drivers/issi/is31fl3741.c#L228,, they're mapped at this point. Is this what's confusing you?

Reply:
https://github.com/melgeek001365/qmk_firmware/pull/1 This might help make things a bit more clear.

Reply:
So I didn't realise I'd made two changes in my suggestion below - as well as renaming the defines, I also renumbered the values so that `CS1_SW1` would be `0x100`. Bit 8 would encode the page number while preserving the original register values as opposed to checking whether the value is >= `0xB4`. This would necessitate two sets of register arrays, so that `g_pwm_buffer_0[0x00]` = CS1_SW1, and `g_pwm_buffer_1[0x00]` = CS1_SW7.

Since none of that has been applied, I think my concern over why there were no logic changes other than to the structure of `is31_led`, and removing `get_cw_sw_position()` in favour of the `CSx_SWx` defines, is rendered moot.